src/ranch_ssl.erl

```diff
@@ -296,3 +297,18 @@ cleanup(#{socket_opts:=SocketOpts}) ->
 	end;
 cleanup(_) ->
 	ok.
+
+-spec strip_usupported_options(opts()) -> opts().
+strip_usupported_options(SocketOpts) ->
+    case lists:keyfind(versions, 1, SocketOpts) of
```

Comment:
I think this is only part of it, ie the `versions` key in the socket options is not the only place influencing it. According to the `ssl` docs [here](http://erlang.org/doc/man/ssl.html#type-protocol_versions), if absent it can also be set by an environment option for the `ssl` application, and if absent there it defaults to all versions that are supported (which can probably be found out by examining the `supported` key in the return from `ssl:versions()`).

Reply:
I don't know if you mean that other options could give a value of `'tlsv1.3'` for a version number, not only `{versions, ['tlsv1.3']}`; however, just to clarify, we found that as long as other options are specified in addition to `'tlsv1.3'`, .e.g. `{versions, ['tlsv1.2', 'tlsv1.3']}`, then the SSL error does not occur.